pandas/_libs/lib.pyx

```diff
@@ -1089,6 +1100,59 @@ def is_float(obj: object) -> bool:
     return util.is_float_object(obj)
 
 
+cpdef bint is_hashable(object obj):
```

Comment:
Does moving this to Cython make a difference for performance? I assume most of the gain is just from short circuiting a hash attempt through the isinstance checks?

Reply:
I've done some benchmarking - but not addressing exact this concern. However the checks for tuple and frozenset will probably be slow in pure Python even using all()

https://github.com/pandas-dev/pandas/pull/59565#issuecomment-2302848473